src/functional/swing_slide_filter.zig

```diff
@@ -210,7 +210,7 @@ pub fn decompress(
     var first_timestamp: usize = 0;
     var index: usize = 0;
     while (index < compressed_lines_and_index.len) : (index += 3) {
-        const current_segment: Segment = .{
+        const current_segment: DiscreteSegment = .{
```

Comment:
Consider looking into if the internals of `decompressSwing()` and `decompressSlide()` overlap enough that it can be moved to a generic function called by `decompressSwing()` and `decompressSlide()`.

Reply:
Ok, I have added a comment with a link to this discussion to #30.